sessiontxn/isolation/readcommitted.go

```diff
@@ -223,7 +223,7 @@ func (p *PessimisticRCTxnContextProvider) AdviseWarmup() error {
 		return nil
 	}
 
-	if err := p.prepareTxn(); err != nil {
+	if err := p.prepareTxn(true); err != nil {
```

Comment:
Is it better to refactor `AdviseWarmup` to only prepareStmtTS when snapshot is disabled because `prepareTxn` will auto consider tidb_snapshot.

```
// AdviseWarmup provides warmup for inner state
func (p *PessimisticRCTxnContextProvider) AdviseWarmup() error {
	if err := p.prepareTxn(true); err != nil {
		return err
	}

	if !p.isTidbSnapshotEnabled() {
		p.prepareStmtTS()
	}
    return nil
}
```

Reply:
Fine.